src/components/dialogs/parameters/dynamicsimulation/curve/common/checkbox-autocomplete.tsx

```diff
@@ -0,0 +1,301 @@
+                </VariableSizeList>
+            </OuterElementContext.Provider>
+        </div>
+    );
+});
+
+const StyledPopper = styled(Popper)({
+    [`& .${autocompleteClasses.listbox}`]: {
+        boxSizing: 'border-box',
+        '& ul': {
+            padding: 0,
+            margin: 0,
+        },
+    },
+});
+
+interface CheckboxAutocompleteProps<T>
+    extends Omit<
+        AutocompleteProps<T, true, false, false, any>,
+        'renderInput' | 'renderOption' | 'inputValue'
+    > {
+    id?: string;
+    virtualize?: boolean;
+    maxSelection?: number;
+    options: T[];
+    getOptionLabel: (option: T) => string;
+    onChangeCallback: (value: T[]) => void;
+}
+
+const CheckboxAutocomplete: React.FC<CheckboxAutocompleteProps<any>> = ({
+    id = '',
+    virtualize = false,
+    maxSelection = 0,
+    options,
+    getOptionLabel,
+    onChangeCallback,
+    ...rest
+}) => {
+    const intl = useIntl();
+
+    // make autocomplete as a controlled component in order to limit maximum selection
+    const [selectedOptions, setSelectedOptions] = useState<(number | string)[]>(
+        []
+    );
+
+    // used to manage search text
+    const [inputValue, setInputValue] = useState<string>('');
+
+    // these states used as conditions to manage input label
+    const [isFocusInput, setIsFocusInput] = useState(false);
+    const [isMaxLimitReached, setMaxLimitReached] = useState(false);
+
+    const debouncedOnChangeCallback = useDebounce(onChangeCallback, 500);
+
+    const handleChange = (_event: React.SyntheticEvent, value: any) => {
+        if (!maxSelection || value.length <= maxSelection) {
+            setMaxLimitReached(false);
+            // cache the selected options
+            setSelectedOptions(value);
+
+            // propagate change to the parent
+            debouncedOnChangeCallback(value);
+        } else {
+            setMaxLimitReached(true);
+        }
+    };
+
+    // when lost focus, show number of options
+    // when focused, check maxSelection reached to show infos
+    const getInputLabel = () => {
+        if (!isFocusInput) {
+            return `${options?.length} ${intl.formatMessage({
+                id: 'options',
+            })}`;
+        } else {
+            if (isMaxLimitReached) {
+                return `${maxSelection} ${intl.formatMessage({
+                    id: 'maxSelection',
+                })}`;
+            }
+        }
+    };
+
+    const renderInput = (params: AutocompleteRenderInputParams) => (
+        <TextField
+            {...params}
+            label={getInputLabel()}
+            onChange={(event) => {
+                setInputValue(event.target.value);
+            }}
+            onFocus={() => {
+                setIsFocusInput(true);
+            }}
+            onBlur={() => {
+                setIsFocusInput(false);
+                setMaxLimitReached(false);
+            }}
+            color={isMaxLimitReached ? 'info' : undefined}
```

Comment:
```suggestion
            color={isMaxLimitReached ? 'warning' : undefined}
```
Not sure but `info` is almost the same color so we should take an other one

Reply:
DONE in https://github.com/gridsuite/gridstudy-app/pull/1836/commits/5ce9528a709c57353a515c819e499b90c240ca6b